library/src/androidTest/java/org/xmtp/android/library/GroupTest.kt

```diff
@@ -0,0 +1,94 @@
+package org.xmtp.android.library
+
+import android.util.Log
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import kotlinx.coroutines.runBlocking
+import org.junit.Assert
+import org.junit.Assert.*
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.xmtp.android.library.messages.PrivateKey
+import org.xmtp.android.library.messages.PrivateKeyBuilder
+import org.xmtp.android.library.messages.walletAddress
+
+@RunWith(AndroidJUnit4::class)
+class GroupTest {
+    lateinit var fakeApiClient: FakeApiClient
+    lateinit var alixWallet: PrivateKeyBuilder
+    lateinit var boWallet: PrivateKeyBuilder
+    lateinit var alix: PrivateKey
+    lateinit var alixClient: Client
+    lateinit var bo: PrivateKey
+    lateinit var boClient: Client
+    lateinit var caroWallet: PrivateKeyBuilder
+    lateinit var caro: PrivateKey
+    lateinit var caroClient: Client
+    lateinit var fixtures: Fixtures
+
+    @Before
+    fun setUp() {
+        val context = InstrumentationRegistry.getInstrumentation().targetContext
+        fixtures =
+            fixtures(clientOptions = ClientOptions(enableLibXmtpV3 = true, appContext = context))
+        alixWallet = fixtures.aliceAccount
+        alix = fixtures.alice
+        boWallet = fixtures.bobAccount
+        bo = fixtures.bob
+        caroWallet = fixtures.caroAccount
+        caro = fixtures.caro
+
+        fakeApiClient = fixtures.fakeApiClient
+        alixClient = fixtures.aliceClient
+        boClient = fixtures.bobClient
+        caroClient = fixtures.caroClient
+    }
+
+    @Test
+    fun testCanCreateAGroup() {
+        val group = boClient.conversations.newGroup(alix.walletAddress)
+        assert(group.id.isNotEmpty())
+    }
+
+    @Test
+    fun testCanListGroupMembers() {
```

Comment:
@nplasterer I figured out what's happening here. There's an error happening, but it's not bubbling up to you.

The error is casued by the fact that caro and alix haven't had `register_identity` called on them yet, so they don't exist on the network.

Will work on bubbling up these kinds of errors better.

Reply:
Hmmm I'm pretty sure I am calling register identity. But I added it again and even with explicit register identity. It still doesn't seem to work, for newGroup or for addMember.

Reply:
I pushed up [a commit](https://github.com/xmtp/xmtp-android/pull/160/commits/43290335f78b92d434548cbd82a439bffdf6feec) that fixes a build issue, but I am seeing some errors in the logs I wouldn't expect and the test also isn't passing for me.

The same case is passing in the [libxmtp tests](https://github.com/xmtp/libxmtp/blob/be2afe8226d3a44fe35d6ccd089354e0a0014fa7/bindings_ffi/src/mls.rs#L573), so something must be getting lost in the FFI layer.

Reply:
Ahh I forgot to push up my local changes that did those things. I'll let you dig into it unless you want a second head. I don't know how helpful I will be debugging the bindings.

Reply:
I found it. It's a casing issue. Once I lowercased the wallet addresses the results started coming back.

We should just standardize all casing at the `libxmtp` level, and none of this should be your problem. I'll put up a PR to do that.

Reply:
O interesting it couldn't find the identity because I was passing in something that was capitalized! Thanks for looking into it!

Reply:
`libxmtp` will handle the casing for you once https://github.com/xmtp/libxmtp/pull/453 goes in